FrameworkSystem/Service/UserProfileManagerHandler.py

```diff
@@ -79,6 +79,32 @@ def export_deleteProfileVar(self, profileName, varName):
     userGroup = credDict['group']
     return gUPDB.deleteVar(userName, userGroup, profileName, varName)
 
+  types_listStatesForWeb = [dict]
+
+  def export_listStatesForWeb(self, permission):
```

Comment:
As of now there's no integration test for the UserProfileClient -> UserProfileHandler -> UserProfileDB (it should be in tests/Integration/Framework). I would say that this is a good occasion for adding it.